hail/src/main/scala/is/hail/sparkextras/MultiWayJoinRDD.scala

```diff
@@ -0,0 +1,52 @@
+package is.hail.sparkextras
+
+import org.apache.spark.{OneToOneDependency, Partition, SparkContext, TaskContext}
+import org.apache.spark.rdd.RDD
+
+import scala.reflect.ClassTag
+
+object MultiWayJoinRDD {
+  def apply[T: ClassTag , V: ClassTag](
+    rdds: IndexedSeq[RDD[T]],
+    preservesPartitioning: Boolean = false
+  )(f: (Array[Iterator[T]]) => Iterator[V]): MultiWayJoinRDD[T, V] = {
+    new MultiWayJoinRDD(rdds.head.sparkContext, rdds, f, preservesPartitioning)
+  }
+}
+
+private class MultiWayJoinPartition[T: ClassTag](
+  idx: Int,
+  @transient private val rdds: Seq[RDD[T]]
```

Comment:
this would be a bit simpler as a case class:

```
case class MultiWayJoinPartition[T : ClassTag](index: Int, partitions: Array[Partition])
```

Then you don't need to worry about transience.